app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -541,6 +557,33 @@ class Result extends React.Component {
     });
   };
 
+  nextSubmission = (direction) => () => {
+    const url = Routes.next_grouping_assignment_submission_result_path(
+      this.state.props.assignment_id, this.state.props.submission_id, this.state.props.result_id
+    );
+
+    $.ajax({
+      url: `${url}?direction=${direction}`
+    }).then((result) => {
+      if (!result.next_result || !result.next_grouping) {
+        alert('There are no more results in this direction!');
+        return;
+      }
+
+      const result_obj = {
+        ...this.state.props,
+        result_id: result.next_result.id,
+        submission_id: result.next_result.submission_id,
+        grouping_id: result.next_grouping.id,
+      };
+      this.setState(prevState => ({ ...prevState, props: result_obj }));
+      let new_url = Routes.edit_assignment_submission_result_url(
+        this.state.props.assignment_id, this.state.props.submission_id, this.state.props.result_id
+      );
+      history.pushState({}, document.title, new_url)
```

Comment:
Nice---does this update the browser URL bar as well?

Reply:
Yep, this updates the browser URL on modern browsers (up-to-date versions of Firefox/Chrome)